lib/default_hostgroup_base_host_patch.rb

```diff
@@ -79,8 +79,8 @@ def host_new_or_forced?
     if Setting[:force_hostgroup_match_only_new]
       # hosts have already been saved during import_host, so test the creation age instead
       new_host = ((Time.current - created_at) < 300)
-      unless new_host && hostgroup.nil? && reports.empty?
-        Rails.logger.debug 'DefaultHostgroupMatch: skipping, host exists'
+      unless new_host && hostgroup.nil? && self.host.reports.empty?
```

Comment:
Like @prasadm1983 suggested this should be:
```suggestion
      unless new_host && self.host.hostgroup.nil? && self.host.reports.empty?
```